Dockerfile

```diff
@@ -0,0 +1,94 @@
+# Prior to building the container, build spark with this command:
+# export MAVEN_OPTS="-Xmx2g -XX:ReservedCodeCacheSize=512m"
+# ./make-distribution.sh --name custom-spark -Dscala-2.11 -Phive -Phive-thriftserver -Phadoop-2.6 -Pyarn -DskipTests -Dgpg.skip=true -Dmaven.javadoc.skip=true
+
+
+# To build the container, run:
+# docker build -t spark:csd-1.6.2 .
+
+FROM ubuntu:trusty
+MAINTAINER ClearStory Data
+
+RUN apt-get update \
+ && apt-get install -y locales \
+ && dpkg-reconfigure -f noninteractive locales \
+ && /usr/sbin/update-locale LANG=C.UTF-8 \
+ && echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen \
+ && locale-gen "en_US.UTF-8" \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Users with other locales should set this in their derivative image
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+RUN apt-get update \
+ && apt-get install -y curl unzip \
+    python3 python3-setuptools \
+ && ln -s /usr/bin/python3 /usr/bin/python \
+ && easy_install3 pip py4j \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update \
+ && apt-get install -y python-software-properties \
+  software-properties-common
+
+RUN add-apt-repository -y ppa:leonard-ehrenfried/apt-transport-s3 \
+ && apt-get update \
+ && apt-get install -y apt-transport-s3
+
+RUN apt-get install -y libsnappy-dev libssl-dev libbz2-dev
+
+# http://blog.stuart.axelbrooke.com/python-3-on-spark-return-of-the-pythonhashseed
+ENV PYTHONHASHSEED 0
+ENV PYTHONIOENCODING UTF-8
+ENV PIP_DISABLE_PIP_VERSION_CHECK 1
+
+# JAVA
+ARG JAVA_MAJOR_VERSION=8
+ARG JAVA_UPDATE_VERSION=92
+ARG JAVA_BUILD_NUMBER=14
+ENV JAVA_HOME /usr/jdk1.${JAVA_MAJOR_VERSION}.0_${JAVA_UPDATE_VERSION}
+
+ENV PATH $PATH:$JAVA_HOME/bin
+RUN curl -sL --retry 3 --insecure \
+  --header "Cookie: oraclelicense=accept-securebackup-cookie;" \
+  "http://download.oracle.com/otn-pub/java/jdk/${JAVA_MAJOR_VERSION}u${JAVA_UPDATE_VERSION}-b${JAVA_BUILD_NUMBER}/server-jre-${JAVA_MAJOR_VERSION}u${JAVA_UPDATE_VERSION}-linux-x64.tar.gz" \
+  | gunzip \
+  | tar x -C /usr/ \
+  && ln -s $JAVA_HOME /usr/java \
+  && rm -rf $JAVA_HOME/man
+
+# HADOOP
+ENV HADOOP_VERSION 2.6.0
+ENV HADOOP_HOME /usr/hadoop-$HADOOP_VERSION
+ENV HADOOP_CONF_DIR=$HADOOP_HOME/etc/hadoop
+ENV PATH $PATH:$HADOOP_HOME/bin
+RUN curl -sL --retry 3 \
+  "http://archive.apache.org/dist/hadoop/common/hadoop-$HADOOP_VERSION/hadoop-$HADOOP_VERSION.tar.gz" \
+  | gunzip \
+  | tar -x -C /usr/ \
+ && rm -rf $HADOOP_HOME/share/doc
+
+RUN ln -nsf $HADOOP_HOME/ /usr/lib/hadoop
+
+# SPARK
+ENV SPARK_VERSION 1.6.2
```

Comment:
Where is this being used?  I can see the need to use "1.6", "1.6.2" or "1.6.2-csd-7" depending on the context.